silo/tasks.py

```diff
@@ -0,0 +1,57 @@
+# Create your tasks here
+from __future__ import absolute_import, unicode_literals
+from celery import shared_task
+from celery.result import AsyncResult
+from tola.celery import app
+
+from random import randint
+from silo.custom_csv_dict_reader import CustomDictReader
+from tola.util import importJSON, saveDataToSilo
+from .models import Silo, Read
+
+import logging
+
+logger = logging.getLogger("tola")
+
+#@shared_task(bind=True, max_retries=3)
+@app.task(bind=True, max_retries=3)
+def async_rand(self, nr):
+    n = randint(0, 100)
+    print("Celery Task Called")
+    return (nr, n)
+
+
+@shared_task(bind=True, retry=False)
+def process_silo(self, silo_id, read_id):
+    silo = Silo.objects.get(id=silo_id)
+    read_obj = Read.objects.get(pk=read_id)
+
+    try:
+        reader = CustomDictReader(read_obj.file_data)
+        saveDataToSilo(silo, reader, read_obj)
+
+        # Todo add notification when done
+        read_obj.task_id = None
+        read_obj.save()
+    except Exception, e:
```

Comment:
Too broad exception clause, isn't it? It's better to catch the exception that you expecting.